app/controllers/user_passwords_controller.rb

```diff
@@ -15,7 +17,7 @@ def create
           respond_with_navigational(resource) { render :new }
         end
         format.js do
-          render json: resource.errors, status: :unauthorized
+          render json: { error: t('email_not_found') }, status: :not_found
```

Comment:
here we're implicitly giving for granted that the only error state is not finding the email, is actually that the only case?

Reply:
Yes, the original code was only responding to that one error, and the views for this controller were hard-coded to respond with the "not found" error message regardless of the error response, but that wasn't previously clear in the controller.